### ISSUE TITLE
Linux/OSX DontLinger not supported in corefx

### ISSUE THREAD

**Issue by raphac** [kerryjiang/SuperSocket#146]
Hello
I use SuperSocket on Linux and have a problem there. On [this](https://github.com/kerryjiang/SuperSocket/blob/f0b85082764bb1307d9dfb9bc6bda4a38ac53c78/SocketEngine/TcpAsyncSocketListener.cs#L45) line, `SocketOptionName.DontLinger` was used and unfortunately that is not supported by Linux. I found and wrote a comment [here](https://github.com/dotnet/corefx/issues/28305#issuecomment-387454356) but nobody answered. That's why I'm trying it here.
I change the code from
`m_ListenSocket.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.DontLinger, true);`
to
`m_ListenSocket.LingerState = new LingerOption(enable: false, seconds: 0);`
and it works. Does the code do the same? If not, what's the best way to solve this problem?

**Comment by chucklu** [comment]
I have checked the file you provided, and the commit is as following:
> commit f0b85082764bb1307d9dfb9bc6bda4a38ac53c78 (origin/v1.6, temp)
> Author: JerYme <jeremy@fizames.name>
> Date:   Wed Feb 7 23:51:53 2018 -0500

However, the corefx version is developed under master branch. And here is a new [listener](https://github.com/kerryjiang/SuperSocket/blob/master/src/SuperSocket.Libuv/LibuvPipeListener.cs) based UvTcpListener  which come from [System.IO.Pipelines.Networking.Libuv](https://dotnet.myget.org/feed/dotnet-corefxlab/package/nuget/System.IO.Pipelines.Networking.Libuv
)

**Comment by raphac** [comment]
Thank you for your answer, but it doesn't solve my problem. I have the [commit](https://github.com/kerryjiang/SuperSocket/commit/f0b85082764bb1307d9dfb9bc6bda4a38ac53c78) in my code, but when I run the application, I get an exception. If I replace the code as described above, then it works, but I do not know what the difference is. I have also seen the new possibility with Libuv, but it is only in the preview and I want to use some features of Supersocket 1.6 (commands, filters...).

**Comment by raphac** [comment]
@chucklu Yes and No :-) I use [SuperSocket-NetCore](https://github.com/SarcoZ/SuperSocket-NetCore) and I made some changes.
